core/services/offchainreporting/peerstore.go

```diff
@@ -16,9 +18,9 @@ import (
 )
 
 type (
-	p2pPeer struct {
-		ID        p2ppeer.ID
-		Addr      ma.Multiaddr
+	P2PPeer struct {
+		ID        string
```

Comment:
these need to be strings because they don't implement Scan/Value?

Reply:
worth using gencodec?

Reply:
Yes

Reply:
Does that mean you're going to update this PR? I'd prefer to have a strong type (ma.Multiaddr, p2ppeer.ID) + gencodec, over a stringly typed struct. But also okay to do in a follow up.

Reply:
Didn't see your comment there JB.

I don't know what gencodec is. Since we're blocked and tests are flakey I'm gonna merge this for now.

Reply:
Do it.